src/components/File.jsx

```diff
@@ -128,4 +130,4 @@ class File extends Component {
   }
 }
 
-export default translate()(File)
+export default withRouter(translate()(File))
```

Comment:
I already ask this: but are you sure `withRouter` is mandatory to have router's functions injected in the component's props? (for `this.props.router.push`)